File: backend/app/schemas/overview.py
from datetime import datetime
from typing import Literal

from pydantic import BaseModel


class DayPoint(BaseModel):
    date: str  # ISO date (YYYY-MM-DD)
    count: int


class TopQuestion(BaseModel):
    question: str
    count: int


class LocationStat(BaseModel):
    location_name: str
    conversation_count: int


class RecentConversation(BaseModel):
    question: str | None
    location_name: str
    started_at: datetime
    status: Literal["resolved", "open"]


class OverviewStatsResponse(BaseModel):
    total_conversations: int
    conversations_delta_pct: float | None

    qr_scans: int
    qr_scans_delta_pct: float | None

    active_locations: int
    total_locations: int

    answer_rate: float | None  # 0-100
    answered_count: int
    question_count: int

    conversations_over_time: list[DayPoint]
    top_questions: list[TopQuestion]
    location_performance: list[LocationStat]
    recent_conversations: list[RecentConversation]

File: frontend/dashboard/src/api/overview.ts
import { api } from "./client";

export interface DayPoint {
  date: string;
  count: number;
}

export interface TopQuestion {
  question: string;
  count: number;
}

export interface LocationStat {
  location_name: string;
  conversation_count: number;
}

export interface RecentConversation {
  question: string | null;
  location_name: string;
  started_at: string;
  status: "resolved" | "open";
}

export interface OverviewStats {
  total_conversations: number;
  conversations_delta_pct: number | null;
  qr_scans: number;
  qr_scans_delta_pct: number | null;
  active_locations: number;
  total_locations: number;
  answer_rate: number | null;
  answered_count: number;
  question_count: number;
  conversations_over_time: DayPoint[];
  top_questions: TopQuestion[];
  location_performance: LocationStat[];
  recent_conversations: RecentConversation[];
}

export function fetchOverviewStats(token: string) {
  return api.get<OverviewStats>("/api/dashboard/overview/stats", token);
}
